src/newrelic/app_test.go

```diff
@@ -184,6 +184,54 @@ func TestConnectPayloadInternalHostname(t *testing.T) {
 	}
 }
 
+func TestConnectPayloadInternalDocker(t *testing.T) {
+	util := &utilization.Data{}
+	info := &AppInfo{}
+
+	// No docker ID, nil utilization data
+	info.DockerId = ""
+
+	b := info.ConnectPayloadInternal(1, nil)
+
+	result, _ := utilization.GetDockerId(b.Util)
```

Comment:
Might be good to test the second (usually ignored) parameter matches expectations.

Reply:
I debated checking that value- `GetDockerId` is just a utility/getter function that I defined specifically just to make getting the value easier by passing a `util` object and returning a `string` and optionally an `error`. The `error` in this case is being ignored, since it's just an error message I came up with for each scenario. I can include checks for it, but it would be testing the `GetDockerId` function moreso than the `ConnectPayloadInternal` function's handling of DockerId, so I left those checks out.

Reply:
Is it reasonable to expect the error is predictable in this test if it succeeds.  If so I think it can't hurt to check it.  If the test fails it would indicate something went wrong with the test or the function itself.  Something I think we'd like to know about.

Reply:
added in 30f4a34c7813a4015cf8417b15edfe7524e7dc01

Reply:
Nice!